store/src/sql/event.rs

```diff
@@ -119,52 +95,47 @@ impl SqliteEventStore {
         rebuild_car(blocks).await
     }
 
-    async fn insert_item_int(&self, item: ReconItem<'_, EventId>) -> Result<(bool, bool)> {
-        let new_val = item.value.is_some();
+    async fn insert_item_int(&self, item: ReconItem<'_, EventId>) -> Result<bool> {
         let res = self.insert_items_int(&[item]).await?;
+        debug!(?res, "insert_item_int");
         let new_key = res.keys.first().cloned().unwrap_or(false);
-        Ok((new_key, new_val))
+        Ok(new_key)
     }
 
     /// Insert many items into the store (internal to the store)
     async fn insert_items_int(&self, items: &[ReconItem<'_, EventId>]) -> Result<InsertResult> {
         if items.is_empty() {
-            return Ok(InsertResult::new(vec![], 0));
+            return Ok(InsertResult::new(vec![]));
         }
         let mut to_add = vec![];
         for item in items {
-            if let Some(val) = item.value {
-                match EventRaw::try_build(item.key.to_owned(), val).await {
-                    Ok(parsed) => to_add.push(parsed),
-                    Err(error) => {
-                        tracing::warn!(%error, order_key=%item.key, "Error parsing event into carfile");
-                        continue;
-                    }
+            match EventRaw::try_build(item.key.to_owned(), item.value).await {
+                Ok(parsed) => to_add.push(parsed),
+                Err(error) => {
+                    tracing::warn!(%error, order_key=%item.key, "Error parsing event into carfile");
+                    continue;
                 }
-            } else {
-                to_add.push(EventRaw::new(item.key.clone(), vec![]));
             }
         }
         if to_add.is_empty() {
-            return Ok(InsertResult::new(vec![], 0));
+            return Ok(InsertResult::new(vec![]));
         }
         let mut new_keys = vec![false; to_add.len()];
-        let mut new_val_cnt = 0;
         let mut tx = self.pool.writer().begin().await.map_err(Error::from)?;
 
         for (idx, item) in to_add.into_iter().enumerate() {
             let new_key = self.insert_key_int(&item.order_key, &mut tx).await?;
-            for block in item.blocks.iter() {
-                self.insert_event_block_int(block, &mut tx).await?;
-                self.mark_ready_to_deliver(&item.order_key, &mut tx).await?;
+            if new_key {
+                // Only add the blocks if this is a new key, otherwise ignore the value.
+                for block in item.blocks.iter() {
+                    self.insert_event_block_int(block, &mut tx).await?;
+                    self.mark_ready_to_deliver(&item.order_key, &mut tx).await?;
```

Comment:
`mark_ready_to_deliver` should be pulled out of the loop, but I have it in the IOD changes I'm working on